src/pages/signup/SignupForm.tsx

```diff
@@ -0,0 +1,279 @@
+                id={CONFIRM_PASSWORD}
+                tag={Field}
+                onChange={(event: React.FormEvent<HTMLInputElement>) => this.handleChange(CONFIRM_PASSWORD, event, setFieldValue)}
+                placeholder="Re-enter password"
+                invalid={!!errors[CONFIRM_PASSWORD] && touched[CONFIRM_PASSWORD]}
+              />
+              <FormFeedback>{errors[CONFIRM_PASSWORD]}</FormFeedback>
+            </FormGroup>
+
+            <FormText className="mb-3" color="danger">
+              {errors[SUBMIT_ERROR] || this.state.providerError}
+            </FormText>
+
+            <Row className="d-flex align-items-center justify-content-end my-5">
+              <Col className="d-flex justify-content-end">
+                <Button
+                  className="btn-primary transition-3d-hover w-100"
+                  type="submit"
+                  onClick={submitForm}
+                  disabled={isSubmitting}
+                  color="primary"
+                >
+                  Create an account
+                </Button>
+              </Col>
+            </Row>
+
+            <Row className="d-flex flex-column align-items-center px-3">
+              <Button
+                className="btn-google transition-3d-hover w-100 d-flex justify-content-between align-items-center"
+                type="button"
+                onClick={this.signInWithProvider.bind(this, signInWithGooglePopUp)}
+              >
+                <i className="fab fa-google" />
+                Signup with Google
+                <div />
+              </Button>
+            </Row>
+
+            <Row className="d-flex align-items-center mt-3">
+              <Col className="d-flex align-items-center justify-content-center">
+                <span className="small text-muted">Already have an account?</span>
+                <Link className="small ml-1" to="/work/login">Login</Link>
+              </Col>
+            </Row>
+          </Form>
+        )}
+      </Formik>
+    );
+  }
+
+  handleSubmit = (values: SignupFormInput, actions: FormikActions<SignupFormInput>) => {
+    const {
+      EMAIL,
+      FIRST_NAME,
+      LAST_NAME,
+      CONFIRM_PASSWORD,
+      SUBMIT_ERROR,
+    } = SignupFormField;
+    const input = {
+      email: values[EMAIL],
+      firstName: values[FIRST_NAME],
+      lastName: values[LAST_NAME],
+      password: values[CONFIRM_PASSWORD],
+    }
+
+    this.props.createUser(input)
+      .then((_) => login(input.email, input.password))
+      .then((_) => !auth.currentUser ? Promise.resolve() : auth.currentUser.sendEmailVerification())
+      .catch((error: Error) => {
+        actions.setErrors({ [SUBMIT_ERROR]: getFriendlyErrorMessage(error) });
+        actions.setSubmitting(false);
+      });
+  };
+
+  handleChange = (htmlId: string, event: React.FormEvent<HTMLInputElement>, setFieldValue: (field: string, value: any) => void) => {
+    setFieldValue(htmlId, event.currentTarget.value);
```

Comment:
can the html id come from the event.currentTarget?

Reply:
nice was able to get it from the `currentTarget.name`~